modules/osdk-golang-implement-controller.adoc

```diff
@@ -0,0 +1,146 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/osdk-golang-tutorial.adoc
+
+[id="osdk-golang-implement-controller_{context}"]
+= Implementing the controller
+
+After creating a new API and controller, you can implement the controller logic.
+
+.Procedure
+
+* For this example, replace the generated controller file `controllers/memcached_controller.go` with the `memcached_controller.go` link:https://github.com/operator-framework/operator-sdk/blob/v1.2.0/testdata/go/memcached-operator/controllers/memcached_controller.go[example implementation].
++
+The example controller executes the following reconciliation logic for each `Memcached` custom resource (CR):
++
+--
+* Create a Memcached deployment if it does not exist.
+* Ensure that the deployment size is the same as specified by the `Memcached` CR spec.
+* Update the `Memcached` CR status with the names of the `memcached` pods.
+--
+
+The next subsections explain how the controller watches resources and how the reconcile loop is triggered. You can skip these subsections to go directly to building and running the Operator.
+
+[id="osdk-golang-implement-controller-resources_{context}"]
+== Resources watched by the controller
+
+The `SetupWithManager()` function in `controllers/memcached_controller.go` specifies how the controller is built to watch a CR and other resources that are owned and managed by that controller.
+
+[source,go]
+----
+import (
+	...
+	appsv1 "k8s.io/api/apps/v1"
+	...
+)
+
+func (r *MemcachedReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&cachev1alpha1.Memcached{}).
+		Owns(&appsv1.Deployment{}).
+		Complete(r)
+}
+----
+
+`NewControllerManagedBy()` provides a controller builder that allows various controller configurations.
+
+`For(&cachev1alpha1.Memcached{})` specifies the `Memcached` type as the primary resource to watch. For each `Memcached` type Add, Update, or Delete event, the reconcile loop is sent a reconcile `Request` argument, a namespace and name key, for that `Memcached` object.
```

Comment:
The second sentence is missing something. PTAL?

Reply:
Awkward sentence casualty of trying to remove a parentheses, IIRC.